core/domain/exp_jobs_one_off.py

```diff
@@ -216,68 +256,133 @@ def map(item):
+    in the exploration.
+
+    This validation is done by the 'is_valid_math_expression' or the
+    'is_valid_math_equation' function present in schema_utils.py.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        is_valid_math_expression = schema_utils.get_validator(
+            'is_valid_math_expression')
+        is_valid_math_equation = schema_utils.get_validator(
+            'is_valid_math_equation')
+        ltt = latex2text.LatexNodes2Text()
+
+        if not item.deleted:
+            exploration = exp_fetchers.get_exploration_from_model(item)
+            for state_name, state in exploration.states.items():
+                if state.interaction.id == 'MathExpressionInput':
+                    new_answer_groups = []
+                    new_interaction_id = ''
+                    for group in state.interaction.answer_groups:
+                        new_answer_group = copy.deepcopy(group)
+                        for rule_spec in new_answer_group.rule_specs:
+                            rule_input = ltt.latex_to_text(
+                                rule_spec.inputs['x'])
+
+                            rule_input = clean_math_expression(rule_input)
+
+                            validity = 'Invalid'
+                            if is_valid_math_expression(rule_input):
+                                validity = 'Valid Algebraic Expression'
+                                new_interaction_id = 'AlgebraicExpressionInput'
+                            elif is_valid_math_expression(
+                                    rule_input, algebraic=False):
+                                validity = 'Valid Numeric Expression'
+                                new_interaction_id = 'NumericExpressionInput'
+                            elif is_valid_math_equation(rule_input):
+                                validity = 'Valid Math Equation'
+                                new_interaction_id = 'MathEquationInput'
+
+                            if validity != 'Invalid':
+                                rule_spec.inputs['x'] = rule_input
+                                if validity == 'Valid Math Equation':
+                                    rule_spec.inputs['y'] = 'both'
+                                rule_spec.rule_type = 'MatchesExactlyWith'
 
                             output_values = '%s %s: %s' % (
                                 item.id, state_name, rule_input)
 
                             yield (validity, output_values.encode('utf-8'))
 
+                        new_answer_groups.append(new_answer_group.to_dict())
+
+                    exp_services.update_exploration(
+                        'admin', item.id, [exp_domain.ExplorationChange({
+                            'cmd': 'edit_state_property',
+                            'state_name': state_name,
+                            'property_name': 'widget_id',
```

Comment:
No changes to customization args?

Also, when an interaction is changed normally through the UI, is this the list of changes that typically results? Please explain in detail what steps you took to verify this.

Reply:
> No changes to customization args?

There are no customization args in the old interaction or in the new one so that won't require changes. 

>  Also, when an interaction is changed normally through the UI, is this the list of changes that typically results? Please explain in detail what steps you took to verify this.

Yes. So, I created two explorations, one with the old interaction and the other with the new one. I then downloaded the explorations in the yaml format and I saw that these are the attributes that are different in both interactions. Also, I have tested this upgrade job locally and it  works as expected.